circuits/core/src/traits.rs

```diff
@@ -14,10 +14,14 @@
 // You should have received a copy of the GNU General Public License
 // along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
 
+use snarkvm_circuits_environment::{Eject, FromBits, Inject, Ternary, ToBits};
+
+use core::fmt::Debug;
+
 /// A trait for a hash function.
 pub trait Hash {
-    type Input;
-    type Output;
+    type Input: Inject + Eject + Ternary + ToBits + FromBits + Clone + Debug;
+    type Output: Inject + Eject + Ternary + ToBits + FromBits + Clone + Debug;
```

Comment:
Can you confirm every trait declared here is necessary?

Reply:
The minimum required seems to be:
```
type Input: Inject + Clone;
type Output: Inject + Eject + Ternary<Output = Self::Output> + ToBits + Clone;
```